application/clicommands/DownloadCommand.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/* Icinga Reporting | (c) 2023 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Reporting\Clicommands;
+
+use InvalidArgumentException;
+use Icinga\Module\Pdfexport\ProvidedHook\Pdfexport;
+use Icinga\Module\Reporting\Cli\Command;
+use Icinga\Module\Reporting\Report;
+
+class DownloadCommand extends Command
+{
+    /**
+     * Download report with specified ID as PDF, CSV or JSON
+     *
+     * USAGE
+     *
+     *     icingacli reporting download <id> [--format=<pdf|csv|json>]
+     *
+     * OPTIONS
+     *
+     * --format=<pdf|csv|json>
+     *     Download report as PDF, CSV or JSON. Defaults to pdf.
+     *
+     * --output=<file>
+     *     Save report to the specified <file>.
+     *
+     * EXAMPLES
+     *
+     * Download report with ID 1:
+     *     icingacli reporting download 1
+     *
+     * Download report with ID 1 as CSV:
+     *     icingacli reporting download 1 --format=csv
+     *
+     * Download report with ID 1 as JSON to the specified file:
+     *     icingacli reporting download 1 --format=json --output=sla.json
+     */
+    public function defaultAction()
+    {
+        $id = $this->params->getStandalone();
+        if ($id === null) {
+            $this->fail($this->translate('Argument id is mandatory'));
+        }
+
+        $output = $this->params->get('output');
+        if ($output === null) {
+            $report = Report::fromDb($id);
+            $format = strtolower($this->params->get('format', 'pdf'));
```

Comment:
Handling `$format` and `$content` is still required to run before the `if`. Now you don't have content if you specify `output`.